[Huang Qing] iP

hqhqhq1 opened the pull request:
# Duke 

> “The way to get started is to quit talking and begin doing.” – Walt Disney [(source)](https://dansilvestre.com/productivity-quotes/#10_Life_is_too_complicated_not_to_be_orderly_-_Martha_Stewart)


### Duke frees your mind of having to remember things you need to do. It's,

- text-based
- very easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/hqhqhq1/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😊

And it is **FREE**!!!!!!😄

   ```
    ____        _        
   |  _ \ _   _| | _____ 
   | | | | | | | |/ / _ \
   | |_| | |_| |   <  __/
   |____/ \__,_|_|\_\___|
   ```
Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```ruby
public static void main(String[] args) throws IOException, ClassNotFoundException {
        new Duke("duke_Saved.txt").run();
}
```

hqhqhq1 committed:
level-1

hqhqhq1 committed:
level-2

Commit:
added Task class and Mark as Done

Commit:
Level 4. ToDos, Events, Deadlines

Commit:
Automated Text UI Testing

Commit:
Level 5. Handle Errors

Commit:
Level-6: Delete

Commit:
branch-Level-7 add readFromFile and writeToFile method

Commit:
Level 8  Dates and Times

Commit:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java
#	text-ui-test/EXPECTED.TXT

Commit:
updated EXPECTED.TXT

Commit:
update input text

Commit:
A-MoreOOP

Commit:
A-Packages

Commit:
A-JUnit

Commit:
A-Jar

hqhqhq1 commented:
v0.1

hqhqhq1 commented:
cannot seem to upload ip.jar file

Commit:
A-JavaDoc

Commit:
A-CodingStandard

Commit:
Level 9 : Find

Commit:
Merge branch 'branch-A-JavaDoc' into branch-Level-9

Commit:
Merge branch 'branch-A-CodingStandard' into branch-Level-9

# Conflicts:
#	src/main/java/Duke/main/Parser.java
#	src/main/java/Duke/main/Ui.java

Commit:
update javadoc for new methods in Ui

Commit:
update Java coding standard

Commit:
Update Ui methods and change a new path to write data

hqhqhq1 committed:
Update README.md

Commit:
A-Gradle

Commit:
A-CheckStyle （will correct all CheckStyle when I have time..)

hqhqhq1 committed:
Update README.md

Commit:
update on checkStyle

Commit:
JavaFX

Commit:
Level-10: GUI, updated Ui class

Commit:
Merge branch 'master' of https://github.com/hqhqhq1/ip

Commit:
update for check style

Commit:
updated GUI

Commit:
added javadoc for getRequest method

Commit:
added javadoc for getResponse method

hqhqhq1 committed:
Update README.md

ckcherry23 commented on src/main/java/Duke/main/Duke.java:
```diff
@@ -0,0 +1,95 @@
+    private Image user = new Image(this.getClass().getResourceAsStream("/images/DaUser.png"));
+    private Image duke = new Image(this.getClass().getResourceAsStream("/images/DaDuke.png"));
+
+    public Duke() {}
```
Is this necessary?

ckcherry23 commented on src/main/java/Duke/main/Duke.java:
```diff
@@ -0,0 +1,95 @@
+        try {
+            tasks = new TaskList(storage.readFromFile());
+        } catch (ClassNotFoundException e) {
+            throw e;
```
Isn't it redundant to catch an exception and throw it again? I believe if you do this, you would lose the stack trace information as well

ckcherry23 commented on src/main/java/Duke/main/Duke.java:
```diff
@@ -0,0 +1,95 @@
+        }
+    }
+
+    /**
+     * give corresponding response for user inputs
+     * @param input read user inputs
+     * @return
+     */
```
I believe javadoc comments require an empty line between the method description and the params/return/throw descriptions. You can also start with a capital letter and end with a period for the method description.

ckcherry23 commented on src/test/java/DeadlineTest.java:
```diff
@@ -0,0 +1,39 @@
+import Duke.task.Deadline;
+import Duke.task.Task;
+import org.junit.jupiter.api.Test;
+
+import java.time.format.DateTimeFormatter;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
```
I think your imports here can be organised/grouped more neatly according to their type?

ckcherry23 commented on src/test/java/DeadlineTest.java:
```diff
@@ -0,0 +1,39 @@
+     * Testing method
+     */
+    @Test
+    public void LocalDateFormatTest() {
```
```suggestion
    public void localDateFormat_validDate_success() {
```
I think you can use the three part format for test method names?

ckcherry23 reviewed (commented):
Looks good! 👍🏼
Just some suggestions and nits to fix.

kev-intq commented on src/main/java/Duke/main/Parser.java:
```diff
@@ -0,0 +1,156 @@
+     */
+    public void userCommand() throws IOException {
+        String inputArr[] = input.split(" ", 2);
+        if (inputArr[0].equals("bye")) {
+            Ui.byeMessage();
+            System.exit(0);
+
+        } else if (inputArr[0].equals("list")) {
+            Ui.listMessage();
+            for (int i = 1; i <= tasks.getTaskList().size(); i++) {
+                System.out.println(i + "." + tasks.getTaskList().get(i - 1));
+            }
+            System.out.println(Ui.LINE);
+
+        } else if (inputArr[0].equals("mark") && isInteger(inputArr[1])) {
+            try {
+                int taskNum = Integer.parseInt(inputArr[1]) - 1;
+                tasks.getTaskList().get(taskNum).markAsDone();
+                Ui.markMessage();
+                System.out.println(tasks.getTaskList().get(taskNum) + "\n" + Ui.LINE);
+                storage.writeToFile(tasks);
+            } catch (IndexOutOfBoundsException exp) {
+                Ui.markErrorMessage();
+            }
```
Good organization of elif statements, the minimal nesting made it very readable. 👍

kev-intq commented on src/main/java/Duke/main/Storage.java:
```diff
@@ -0,0 +1,50 @@
+package Duke.main;
+
+
```
There seems to be an extra line here

kev-intq commented on src/main/java/Duke/main/Ui.java:
```diff
@@ -0,0 +1,133 @@
+     */
+    public static void generalErrorMessage() {
+        System.out.println(LINE + "☹ OOPS!!! I'm sorry, but I don't know what that means :-( \n"
+                + "To view all task available: list\n"
+                + "To add more task: todo ...   deadline .../...    event .../...  or mark/unmark taskNumber\n"
+                + "To delete any task available: delete taskNumber\n"
+                + "please try again\n" + LINE);
```
Good indentation and following of line break and wrap rules.

Commit:
Update parser class userCommand() method,

user input can be Uppercase in any letter, covert the commend to lower case for correct execution

update on assertions

hqhqhq1 committed:
Merge pull request #1 from hqhqhq1/A-Assertions

Update parser class userCommand() method,

Commit:
Parser class refactoring using extract Method

userCommand() had all the logic written

Extracting the different commands and created new classes for all the different commands

calling the corresponding user input command in userCommand()

hqhqhq1 committed:
Merge pull request #2 from hqhqhq1/heads/A-CodeQuality

Parser class refactoring using extract Method

Commit:
A-CI

Commit:
Parser class: Added help command for command format guide

Duke class: remove unused parameter

Ui class: updated generalErrorMessage

Storage class: updated  check style

Commit:
Updated on GUI
change pictures and added background picture
update UI greeting message

Commit:
added Ui.png

hqhqhq1 committed:
Set theme jekyll-theme-cayman

hqhqhq1 committed:
Update README.md

hqhqhq1 committed:
Update README.md

Commit:
change main class in build.gradle

Commit:
fix Duke constructor

Commit:
fix jar cannot read file bug

Commit:
update MainWindow gui

hqhqhq1 committed:
Update README.md
